From 46b21606e63ed29c2d93029296a91c6dc513da24 Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Mon, 9 Nov 2020 11:54:58 +0100
Subject: [PATCH] kerneldevice,generic: plug cached string leaks

Both 'physdev_subsystem' and 'interface_description' were leaking.

(cherry picked from commit 37b3771ef5003adb1797c29deceb6500ebe60e6d)
---
 src/kerneldevice/mm-kernel-device-generic.c | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/src/kerneldevice/mm-kernel-device-generic.c b/src/kerneldevice/mm-kernel-device-generic.c
index 254c505a..3349bd46 100644
--- a/src/kerneldevice/mm-kernel-device-generic.c
+++ b/src/kerneldevice/mm-kernel-device-generic.c
@@ -1046,13 +1046,15 @@ dispose (GObject *object)
 {
     MMKernelDeviceGeneric *self = MM_KERNEL_DEVICE_GENERIC (object);
 
-    g_clear_pointer (&self->priv->physdev_product,      g_free);
-    g_clear_pointer (&self->priv->physdev_manufacturer, g_free);
-    g_clear_pointer (&self->priv->physdev_sysfs_path,   g_free);
-    g_clear_pointer (&self->priv->interface_sysfs_path, g_free);
-    g_clear_pointer (&self->priv->sysfs_path,           g_free);
-    g_clear_pointer (&self->priv->driver,               g_free);
-    g_clear_pointer (&self->priv->rules,                g_array_unref);
+    g_clear_pointer (&self->priv->physdev_product,       g_free);
+    g_clear_pointer (&self->priv->physdev_manufacturer,  g_free);
+    g_clear_pointer (&self->priv->physdev_subsystem,     g_free);
+    g_clear_pointer (&self->priv->physdev_sysfs_path,    g_free);
+    g_clear_pointer (&self->priv->interface_description, g_free);
+    g_clear_pointer (&self->priv->interface_sysfs_path,  g_free);
+    g_clear_pointer (&self->priv->sysfs_path,            g_free);
+    g_clear_pointer (&self->priv->driver,                g_free);
+    g_clear_pointer (&self->priv->rules,                 g_array_unref);
     g_clear_object  (&self->priv->properties);
 
     G_OBJECT_CLASS (mm_kernel_device_generic_parent_class)->dispose (object);
-- 
2.25.1

